cookbook/ESPHome-IR-amplifier-control.rst

```diff
@@ -0,0 +1,205 @@
+ESPHome IR HiFi amplifier control
+=================================
+
+.. seo::
+    :description: The goal is to remote control the volume, a standard knob managing a potentiometer, and also switch the HiFi amplifier on and off.
+    :image: ESPHome-IR-amplifier-control.jpg
+    :keywords: IR servo mqtt ESPHome
+
+The goal is to remote control the volume, a standard knob managing a potentiometer, and also switch the HiFi amplifier on and off.
+
+The project is based on an ESP32 and ESPHome, to receive and decode the IR from a remote control. It can use any remote with a few minor adjustements.
+
+For the ON/OFF the project uses a Shelly Plug S managed via web from Node-RED. One of the keys of the remote is decoded and triggers a MQTT message that Node-RED can capture and convert to a relay toggle order for the Shelly Plug. Of course any other smart plug can be used if it accepts commands via web.
+
+.. figure:: images/ESPHome-IR-amplifier-control.jpg
+    :align: center
+    :width: 50.0%
+
+The servo and the board are fixed with double sided tape, so everything is removable.
+
+.. figure:: images/apple_remote.jpg
+    :align: left
+    :width: 100.0%
+
+The remote used
+
+.. figure:: images/servo_pulley.jpg
+    :align: left
+    :width: 100.0%
+
+and the servo pulley (`servo_pulley.stl <https://github.com/deltazerorsan/ESPHome-IR-HiFi-amplifier-control/blob/master/ESPHome_yaml/irreceiver.yaml>`__)
+
+So, the project needs:
+
+- TSOP38238 to receive infrared signals
+- ESP-WROOM-32 module (or any board with similar module)
+- 360 degrees small servo (not a standard one, but 360 degrees)
+- Appropriate belt
+- some 3.3 power source
+  - 5 volts to 3.3 converter (it can be onboard)
+  - Wall 5 volts power supply (or something equivalent)
+- Any smart plug like Shelly Plug S or similar
+
+Infrastructure:
+
+- WiFi network
+- ESPHome running (in my case Hassio version)
+- Node-RED running (in my case Hassio version) (`Download node here in JSON to import to Node-RED <https://github.com/deltazerorsan/ESPHome-IR-HiFi-amplifier-control/blob/master/ESPHome_yaml/ir_amplifier_control.ndr>`__)
+
+
+This is the yaml code: (`Download irreceiver.yaml from here <https://github.com/deltazerorsan/ESPHome-IR-HiFi-amplifier-control/blob/master/ESPHome_yaml/irreceiver.yaml>`__)
+
+.. code-block:: yaml
+
+    substitutions:
+      devicename: irreceiver
+      uper_devicename: IR Receiver
+      device_ip: 192.168.1.33
+    
+    esphome:
+      name: $devicename
+      platform: ESP32
+      board: esp-wrover-kit
+    
+      on_boot:
+        priority: -10
+        then:
+          - servo.write:
+              id: my_servo
+              level: 0                   # put the servo in neutral (stoped)
+          - delay: 100ms
+          - servo.detach: my_servo       # always detach the servo to ensure it stops completely

```

Comment:
this is a bug currently in v1.14, should work ok in next version (dev) you might want to review this when updating

Reply:
Ok, I will try when it is solved.